src/Orleans.KafkaStreamProvider/KafkaQueue/KafkaStreamProviderOptions.cs

```diff
@@ -164,5 +169,23 @@ private static bool GetOptionalParamBool(string paramName, bool defaultValue, IP
 
             return paramValue;
         }
+        
+        private Func<IKafkaBatchFactory> GetOptionalParamKafkaBatchFactory(string paramName, Func<IKafkaBatchFactory> defaultValue, IProviderConfiguration config, IServiceProvider serviceProvider)
```

Comment:
Can you add test with custom IKafkaBatchFactory that use other sterilization like json.net .